src/components/SegmentControl/CollectionSegmentControl.js

```diff
@@ -1,31 +1,47 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { View, StyleSheet } from 'react-native';
 import { connect } from 'react-redux';
-import { bool, func } from 'prop-types';
+import { bool, func, shape } from 'prop-types';
 import BaseSegmentControl from '../Generic/BaseSegmentControl';
 
 import BaseText from '../Generic/BaseText';
 import { toggleShowCollectionOnly } from '../../redux/action-creators';
+import { filterTriggerDateRangeSelector } from '../../redux/selectors';
+import { actionTypes } from '../../redux/action-types';
 
 const allRecordsDescription = 'Displays all records.';
 const collectionRecordsDescription = 'Only displays records saved to the collection.';
 
 const CollectionSegmentControl = ({
   showCollectionOnly,
   toggleShowCollectionOnlyAction,
+  updateDateRangeFilter,
+  filterTriggerDateRange,
+  hasCollectionIds,
 }) => {
   const segControlIndex = showCollectionOnly ? 1 : 0;
   const description = segControlIndex === 0 ? allRecordsDescription : collectionRecordsDescription;
   const handleChange = (selectedSegmentIndex) => {
     toggleShowCollectionOnlyAction(selectedSegmentIndex !== 0);
+    updateDateRangeFilter(filterTriggerDateRange);
   };
 
+  // reset SegmentControl and TimelineRange when user
+  // clears Collection Records while in Show Collection Only view
+  useEffect(() => {
+    if (showCollectionOnly && !hasCollectionIds) {
+      toggleShowCollectionOnlyAction(false);
+      updateDateRangeFilter(filterTriggerDateRange);
```

Comment:
this may not be necessary, if all primary values necessary to update the date range are already in redux state.  (including `howCollectionOnly` and `hasCollectionIds`)

...the daterange filter already responds to available records.  (only currently, they don't respond to ^these filters.)

...it might be better to adjust daterange values in a follow-up PR, without dispatching an action or using `useEffect`